travelwise/api/views.py

```diff
@@ -224,3 +222,14 @@ def validateUser(request, username=None):
         
         return JsonResponse({"message": "good"})
 
+
+@login_required(login_url="landing")
+def delete_plan(request, plan_id):
+    if request.method == 'GET':
+        try:
+            TravelPlan.objects.get(pk=plan_id, author=request.user).delete()
+        except Exception as err:
+            return JsonResponse({'error': 'Plan was not deleted'}, status=HTTPStatus.BAD_REQUEST)
+
+        return redirect("history")
```

Comment:
You should either design an endpoint to be machine readable, or human readable. You shouldn't mix both.

Reply:
Fixed. Using django messages to add the error messages, so only redirecting at the end.